x-pack/plugins/timelines/server/search_strategy/index_fields/index.test.ts

```diff
@@ -231,146 +231,199 @@ describe('Fields Provider', () => {
-    });
-
-    it('should check index exists', async () => {
-      const indices = ['some-index-pattern-*'];
-      const request = {
-        indices,
-        onlyCheckIfIndicesExist: true,
-      };
-
-      const response = await requestIndexFieldSearch(request, deps, beatFields, getStartServices);
-      expect(response.indexFields).toHaveLength(0);
-      expect(response.indicesExist).toEqual(indices);
-    });
-
-    it('should search index fields', async () => {
-      const indices = ['some-index-pattern-*'];
-      const request = {
-        indices,
-        onlyCheckIfIndicesExist: false,
-      };
-
-      const response = await requestIndexFieldSearch(request, deps, beatFields, getStartServices);
-
-      expect(getFieldsForWildcardMock).toHaveBeenCalledWith({ pattern: indices[0] });
-
-      expect(response.indexFields).not.toHaveLength(0);
-      expect(response.indicesExist).toEqual(indices);
-    });
-
-    it('should search index fields by data view id', async () => {
-      const dataViewId = 'id';
-      const request = {
-        dataViewId,
-        onlyCheckIfIndicesExist: false,
-      };
-
-      const response = await requestIndexFieldSearch(request, deps, beatFields, getStartServices);
-
-      expect(getFieldsForWildcardMock).not.toHaveBeenCalled();
-
-      expect(response.indexFields).not.toHaveLength(0);
-      expect(response.indicesExist).toEqual(['coolbro']);
-    });
-
-    it('onlyCheckIfIndicesExist by data view id', async () => {
-      const dataViewId = 'id';
-      const request = {
-        dataViewId,
-        onlyCheckIfIndicesExist: true,
-      };
-
-      const response = await requestIndexFieldSearch(request, deps, beatFields, getStartServices);
-
-      expect(response.indexFields).toHaveLength(0);
-      expect(response.indicesExist).toEqual(['coolbro']);
-    });
-
-    it('should search apm index fields', async () => {
-      const indices = ['apm-*-transaction*', 'traces-apm*'];
-      const request = {
-        indices,
-        onlyCheckIfIndicesExist: false,
-      };
-
-      const response = await requestIndexFieldSearch(request, deps, beatFields, getStartServices);
-
-      expect(getFieldsForWildcardMock).toHaveBeenCalledWith({ pattern: indices[0] });
-      expect(response.indexFields).not.toHaveLength(0);
-      expect(response.indicesExist).toEqual(indices);
-    });
+    const depsInternalESUser = {
+      esClient: {
+        asInternalUser: { search: esClientSearchMock, fieldCaps: esClientFieldCapsMock },
+      },
+    } as unknown as SearchStrategyDependencies;
 
-    it('should check apm index exists with data', async () => {
-      const indices = ['apm-*-transaction*', 'traces-apm*'];
-      const request = {
-        indices,
-        onlyCheckIfIndicesExist: true,
-      };
+    describe.each([
+      ['currentESUser', depsCurrentESUser, false],
+      ['internalESUser', depsInternalESUser, true],
+    ])(`Using %s`, (_, deps, useInternalUser) => {
```

Comment:
nit: instead of `useInternalUser`, maybe rename this param to just `user` or something

Reply:
It is a boolean to determine if the test uses the internal user or the current user for ES search, `user` seems to me that it contains a user object in this case, which is not the case, what do you think?

Reply:
of course. I just suggested so that on a quick read it is not confusing to see internalUser for all the tests 😅